### ISSUE TITLE
torch.tensor(numpy.float64()) creates a float32 tensor

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#27754]
## 🐛 Bug

```
>>> import torch
>>> import numpy

>>> t = torch.tensor(numpy.float64())
>>> t.dtype
torch.float32
```

Should be torch.float64. 

test_dataloader.py has test_numpy_scalars which was supposed to test for this, but that test historically ran with the default tensor dtype set to torch.double and that masked this issue. In #27628 I removed setting the default tensor dtype and discovered the error. The corresponding line of the test is commented out pending resolution of this issue. 

cc @ezyang @SsnL @gchanan @zou3519

**Comment by gchanan** [comment]
I don't think we ever handled the dtype scalars correctly, as opposed to the tensors.

We should though.

**Comment by gchanan** [comment]
ya, same thing happens in 1.2.

**Comment by mruberry** [comment]
I agree it's likely not a regression, but other numpy scalars (like numpy.float16()) do work properly and the double behavior was tested for. The other dtype tests still pass.

**Comment by gchanan** [comment]
I think this is because a `np.float64` is an instance of `PyFloat` so it enters that codepath (and thus you get the default dtype behavior).

Honestly any choice here seems non-obvious:
- "I asked for float64 but you gave me float32!" (if we use default dytpe)
- "If I pass a float64, I get a float64, but if I take the python value (which is exactly equivalent) I get a float32!" (if we don't use default dtype)

**Comment by albanD** [comment]
But why is `PyFloat` converting to `float32`? Python floats are double precision no?

**Comment by gchanan** [comment]
Because you'd always type infer float64s if you did that.  E.g. `torch.tensor(1.)` would give you a float64.  That's not usually what you want, so we give the default_dtype as a setting.

**Comment by gchanan** [comment]
We think this should give you a float64, but this would be BC breaking, so we need to document, etc.

**Comment by nairbv** [comment]
I think this was fixed by https://github.com/pytorch/pytorch/pull/30486

```
>>> import numpy
>>> import numpy as np
>>> t=torch.tensor(numpy.float64())
>>> t.dtype
torch.float64
>>> t=torch.tensor(numpy.float32())
>>> t.dtype
torch.float32
```

**Comment by gchanan** [comment]
@nairbv can you check if it's fixed and close this issue if so?

**Comment by nairbv** [comment]
looks fixed to me (based on example given)
